Rename `setfrozentag` action to `set_milestone_frozen_marker`

AliSoftware opened the pull request:
## What does it do?

Rename `setfrozentag` action to `set_milestone_frozen_marker`, And also make sure it can handle milestone with custom titles (ones that have more than just the version name in their title but also custom text).

Also adds unit tests on the action (there weren't any).

## Why?

Because the name of that action was unclear and not following our conventions:
 - The name of the action, as it would be used in our `Fastfile`, was compacted in a single word without spaces (aka `_` in camel_case), i.e. `setfrozentag`, vs `set_frozen_tag`, making it not super readable
 - The naming and wording was not ideal either, as one could easily think that this action was related to GitHub tags, and nothing indicated that this was related to GitHub milestones instead, and had nothing to do with git tags.

Since we're approaching a new major release (as we have Breaking Changes in our changelog waiting to be released), I thought this would be the perfect occasion to clean this one up while at it.

## Checklist before requesting a review

- [x] Run `bundle exec rubocop` to test for code style violations and recommendations
- [x] Add Unit Tests (aka `specs/*_spec.rb`) if applicable
- [x] Run `bundle exec rspec` to run the whole test suite and ensure all your tests pass
- [x] Make sure you added an entry in [the `CHANGELOG.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/CHANGELOG.md#trunk) to describe your changes under the appropriate existing `###` subsection of the existing `## Trunk` section.
- [x] If applicable, add an entry in [the `MIGRATION.md` file](https://github.com/wordpress-mobile/release-toolkit/blob/trunk/MIGRATION.md) to describe how the changes will affect the migration from the previous major version and what the clients will need to change and consider.

AliSoftware committed:
Rename `setfrozentag` action to `set_milestone_frozen_marker`

And also make sure it can handle milestone with custom titles (ones that have more than just the version name in their title but also custom text)

dangermattic commented:
<!--
  0 Errors
  3 Warnings: Please add an entry in the CHA...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          3 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Please add an entry in the CHANGELOG.md file to describe the changes made by this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">No reviewers have been set for this PR yet. Please request a review from <strong>@⁠wordpress-mobile/apps-infrastructure</strong>.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

AliSoftware committed:
Add Unit Tests for `set_milestone_frozen_marker`

AliSoftware committed:
Add CHANGELOG and MIGRATION entries

AliSoftware commented on lib/fastlane/plugin/wpmreleasetoolkit/actions/common/set_milestone_frozen_marker_action.rb:
```diff
@@ -12,23 +12,28 @@ def self.run(params)
-          mile_title = milestone_title
+          unless is_frozen(milestone)
+            UI.message("Milestone `#{mile_title}` is not frozen. Nothing to do")
+            return # Already frozen: nothing to do
```
```suggestion
            return # Not frozen: nothing to do
```

AliSoftware committed:
Fix typo in comment

mokagio commented on lib/fastlane/plugin/wpmreleasetoolkit/actions/common/set_milestone_frozen_marker_action.rb:
```diff
@@ -52,7 +57,7 @@ def self.return_value
 
       def self.details
         # Optional:
```
Nitpick: I guess the Fastlane template has this comment to let people know they don't have to add one, but given we added it, might as well remove it.

```suggestion
```

Or, since it's exactly the same as `description`, cash into it being optional and omitting it.

AliSoftware replied:
Addressed in 80ea60473c1b52e23d25c4c1971626ec3298b1b4

mokagio commented on spec/set_milestone_frozen_marker_action_spec.rb:
```diff
@@ -0,0 +1,80 @@
+    allow(Octokit::Client).to receive(:new).and_return(client)
+  end
+
+  context 'when passing `freeze: false` (default)' do
```
```suggestion
  context 'when passing `freeze: true` (default)' do
```

AliSoftware replied:
Nice catch, thanks 👍

mokagio reviewed (approved):
All good to merge functionality wise, but there's a typo in the test descriptions that ought to be addressed for the sake of logs clarity.

AliSoftware committed:
Fix typo in spec description

Co-authored-by: Gio Lodi <gio.lodi@automattic.com>

AliSoftware committed:
Remove unused methods from action template

As suggested by @mokagio in https://github.com/wordpress-mobile/release-toolkit/pull/548/files#r1503466813

AliSoftware committed:
DRY some stubs in specs

AliSoftware committed:
Merge branch 'trunk' into rename-setfrozentag

# Conflicts:
#	CHANGELOG.md
#	MIGRATION.md
